spec/rspec/core/hooks_filtering_spec.rb

```diff
@@ -319,6 +319,23 @@ module RSpec::Core
         expect(filters).to eq([])
       end
 
+      it "does not run :all|:context hooks if the entire context is skipped" do
+        filters = []
+        RSpec.configure do |c|
+          c.before(:all) { filters << "before all in config"}
+          c.after(:all) { filters << "after all in config"}
+          c.before(:context) { filters << "before context in config"}
+          c.after(:context) { filters << "after context in config"}
+        end
+        group = RSpec.xdescribe("skipped describe") do
+          xcontext("skipped context") do
+            it("is skipped") {}
+          end
+        end
```

Comment:
I just realized that this spec is only covering global config hooks, but does not cover `:context` hooks declared in the skipped example group itself.  Given that in this case we want _all_ `:context` hooks to be skipped, regardless of where they are declared, it would be good to cover that in this spec.  Mind improving the spec before we merge?